pkg/kubectl/cmd/create_deployment.go

```diff
@@ -36,10 +37,25 @@ var (
 	kubectl create deployment my-dep --image=busybox`))
 )
 
+type DeploymentOpts struct {
+	CreateSubcommandOptions *CreateSubcommandOptions
+
+	CmdErr io.Writer
+}
+
 // NewCmdCreateDeployment is a macro command to create a new deployment.
 // This command is better known to users as `kubectl create deployment`.
 // Note that this command overlaps significantly with the `kubectl run` command.
 func NewCmdCreateDeployment(f cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Command {
+	options := &DeploymentOpts{
+		CreateSubcommandOptions: &CreateSubcommandOptions{
+			PrintFlags: printers.NewPrintFlags("created"),
+			CmdOut:     cmdOut,
+		},
+
+		CmdErr: cmdErr,
```

Comment:
Put `CmdErr` into the common struct `CreateSubcommandOptions` even if not all use it, it still does make sense to be just there.